src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Microsoft.Android.Sdk.Bindings.Gradle.targets

```diff
@@ -0,0 +1,93 @@
+<!--
+***********************************************************************************************
+Microsoft.Android.Sdk.Bindings.Gradle.targets
+
+This file contains MSBuild targets that support building and operating on Android Gradle projects.
+
+***********************************************************************************************
+-->
+
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <UsingTask TaskName="Xamarin.Android.Tasks.Gradle"              AssemblyFile="Xamarin.Android.Build.Tasks.dll" />
+
+  <PropertyGroup>
+    <_BuildAndroidGradleProjectsStamp Condition=" '$(_BuildAndroidGradleProjectsStamp)' == '' ">$(IntermediateOutputPath)stamp/_BuildAndroidGradleProjects.stamp</_BuildAndroidGradleProjectsStamp>
+  </PropertyGroup>
+
+  <ItemDefinitionGroup>
+    <AndroidGradleProjectReference>
+      <Bind>true</Bind>
+      <Configuration>Release</Configuration>
+      <ReferenceLibraryOutputs>true</ReferenceLibraryOutputs>
+      <ModuleName></ModuleName>
+      <OutputPath>$(MSBuildProjectDirectory)/$(OutputPath)gradle/</OutputPath>
+      <Pack>true</Pack>
+      <Visible>false</Visible>
+    </AndroidGradleProjectReference>
+  </ItemDefinitionGroup>
+
+
+  <Target Name="_GetBuildAndroidGradleProjectsInputs">
+    <ItemGroup>
+      <_GradleInputs Include="$(MSBuildProjectFullPath)" />
+      <_GradleInputs Include="%(AndroidGradleProjectReference.FullPath)/**/*.java" />
+      <_GradleInputs Include="%(AndroidGradleProjectReference.FullPath)/**/*.kt" />
+      <_GradleInputs Include="%(AndroidGradleProjectReference.FullPath)/**/*.kts" />
+      <_GradleInputs Include="%(AndroidGradleProjectReference.FullPath)/**/*.gradle" />
+      <_GradleInputs Include="%(AndroidGradleProjectReference.FullPath)/**/*.properties"/>
+      <_GradleInputs Include="%(AndroidGradleProjectReference.FullPath)/**/*.xml" />
```

Comment:
Will this include the gradle equivelents of `obj` in the search ? becuase gradle does produce allot of intermediate files, it might cause some slow evaluations?

Reply:
Only if the `OutputPath` metadata is set by the customer to a path under the Gradle project. The default build path is in the output path of the .NET project, but I think I can update this to exclude content in `%(AndroidGradleProjectReference.OutputPath)` in case it's overridden.

Reply:
All the wildcards seem like they could be slow like I saw in:
* https://github.com/dotnet/maui/pull/24453

If you have a `.binlog`, I wonder how much time it takes as it would run these on every build.

The current code looks like it would run `**` in that folder 6 times. Is there a way it could be nonrecursive, or have less wildcards?

Reply:
I've updated this to do the following instead, but still need to do some perf measuring:
```xml
<_AllGradleFiles Include="$([System.IO.Directory]::GetFiles('%(AndroidGradleProjectReference.FullPath)', '*.*', System.IO.SearchOption.AllDirectories))"
    Condition= " Exists('%(AndroidGradleProjectReference.FullPath)') "/>
<_GradleInputs Include="@(_AllGradleFiles)"
    Condition="  '%(Extension)' == '.java'
              or '%(Extension)' == '.kt'
              or '%(Extension)' == '.kts'
              or '%(Extension)' == '.gradle'
              or '%(Extension)' == '.properties'
              or '%(Extension)' == '.xml' " />
```

Reply:
I've updated this again to use a single wildcard glob which is then filtered. The initial glob is conditional based on the existence of the directory being searched so we don't accidentally or arbitrarily scan a bunch of directories.